broker-daemon/orderbook/index.spec.js

```diff
@@ -312,17 +313,88 @@ describe('Orderbook', () => {
         expect(watcher.removeListener).to.have.been.calledWith('error', sinon.match.func)
       })
 
-      it('sets a 5 second timeout', () => {
+      it('exponentially backs off after repeated end events', () => {
         expect(timeoutStub).to.have.been.calledOnce()
-        expect(timeoutStub).to.have.been.calledWith(sinon.match.func, 5000)
+        expect(timeoutStub).to.have.been.calledWith(sinon.match.func, 1000)
+
+        const expectedResults = [3000, 7000, 15000, 31000]
+
+        for (let i = 0; i < expectedResults.length; i++) {
+          onEnd()
+          const expected = expectedResults[i]
+          expect(timeoutStub.lastCall).to.have.been.calledWith(sinon.match.func, expected)
+        }
+
+        // Test the MAX_RETRY_INTERVAL is reached and maintained
+        for (let i = 0; i < 2; i++) {
+          onEnd()
+          const expected = 60000
+          expect(timeoutStub.lastCall).to.have.been.calledWith(sinon.match.func, expected)
+        }
       })
 
       it('re-initializes after the timeout', () => {
         const timeoutFunc = timeoutStub.args[0][0]
+        const updatedRetries = timeoutStub.args[0][2]
```

Comment:
what is this parameter? I don't think there is a 3rd parameter passed to `setTimeout` anymore, is there?